lib/net.js

```diff
@@ -834,6 +834,15 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
+// Check that the port number is not NaN when coerced to a number,
+// is an integer and that it falls within the legal range of port numbers.
+function isLegalPort(port) {
+  if (typeof port === 'string' && port.trim() === '')
+    return false;
```

Comment:
Okay, that much I get.  It's not really clear to me though why undefined is allowed (and interpreted as 0) but not a blank string.